### ISSUE TITLE
Problem trying to receive data from parse using react.

### ISSUE THREAD

**Issue by BrunoAl** [ParsePlatform/ParseReact#174]
Hello,
I have connected react with parse, following this: https://github.com/ParsePlatform/ParseReact, but I can't receive data using the observe(), the chrome's console panel tells me this: "Uncaught TypeError: Cannot read property 'pin' of undefined". When I try to send data using ParseReact.Mutation, everything works fine.
My code look like this: 
import React from 'react';
import Parse from 'parse';
import ParseReact from 'parse-react';

import LocalePinRegister from './LocalePinRegister.jsx';
import EventPinReginter from './EventPinRegister.jsx';
import SubmitButton from './SubmitButton'
const pinRegister = React.createClass({
  mixis: [ParseReact.Mixin],

  observe() {
    return {
      pins: (new Parse.Query('Pin'))
    };
  },
  getInitialState() {...}

  render: function() {
    return (
      <ul>
        {this.data.pins.map(function(pin) {
          return <li>{pin}</li>;
        })}
      </ul>
    );
  }

I've trying for a whole week to figure this out, but a couldn't find the problem. Can someone help me?

**Comment by moflo** [comment]
Hi, seeing this same issue when running a Parse server on Heroku. Thought it was a problem with their latest stack, but it seems they fully support WS [now](https://devcenter.heroku.com/articles/node-websockets). Any suggestions?

**Comment by BrunoAl** [comment]
Yes, I realized that I was missing the liveQuery on my node server, and the socket error stopped,
```
var api = new ParseServer({
  databaseURI: databaseUri || 'mongodb://54.94.198.98:23988',
  cloud: process.env.CLOUD_CODE_MAIN || __dirname + '/cloud/main.js',
  appId: process.env.APP_ID || 'myAppId',
  masterKey: process.env.MASTER_KEY || 'spen', //Add your master key here. Keep it secret!
  serverURL: process.env.SERVER_URL || 'http://localhost:1337',  // Don't forget to change to https if needed
  facebookAppIds: ["598082443687853"],
  javascriptKey: '7t55tgyv666',
  liveQuery: {
    classNames: ['Pin'']
  }
});
```
But now now I'm still not enable to receive on chrome's console I get this: Uncaught TypeError: this._subscriptions[name].dispose is not a function, and this problem comes from this peace of code from parse-react:
 ```
_unsubscribe: function _unsubscribe() {
    for (var name in this._subscriptions) {
      this._subscriptions[name].dispose();
    }
    this._subscriptions = {};
  },
```

**Comment by moflo** [comment]
I too missed that LiveQuery was a requirement (!) and am now also able to POST but not receive any data via web sockets / LiveQuery observe methods. FWIW, I'm using the following:

      "parse": "~1.8.3",
      "parse-server": "~2.2.7"
      "parse-react": "^0.5.1",

**Comment by BrunoAl** [comment]
Anyone have any idea of how to solve this? It's a major issue, and it looks like nobody is talking about this on the web. On my developer team, the android dev had no problem with parse using the exactly same nodejs server, so it must be something wrong on parse-react.

**Comment by dcalogue** [comment]
Could you try the following? 
Add the following logs to the ParsePatches.js file

`
var ParsePatches = {
  applyPatches: function applyPatches() {
    console.log("Applying parse react patches");
    if (!Parse.Object.prototype.toPlainObject) {
      Parse.Object.prototype.toPlainObject = patches.toPlainObject;
    }
    if (!Parse.Query.prototype.subscribe) {
      console.log("Applying suscribe to Query");
      Parse.Query.prototype.subscribe = patches.subscribe;
    }
    if (!Parse.Query.prototype.observeOne) {
      Parse.Query.prototype.observeOne = patches.observeOne;
    }`

It seems that someone introduced a suscribe function into the Parse code, which breaks the Parse patch from ParseReact (if you want both ParseReact and liveQuery you should change all references to subscribe in the ParseReact code to another function name).

Let me know if this fixes your problem (I had a similar problem and had to dive into the code to find what was going on)

**Comment by BrunoAl** [comment]
@dcalogue I think you are! I was too lazy to change all the subscribe's reference name, so I did this just to give a shot.
```
var ParsePatches = {
  applyPatches: function applyPatches() {
    console.log("Applying parse react patches");
    if (!Parse.Object.prototype.toPlainObject) {
      Parse.Object.prototype.toPlainObject = patches.toPlainObject;
    }
    if (true) {
      console.log("Applying subscribe to Query");
      Parse.Query.prototype.subscribe = patches.subscribe;
    }
    if (!Parse.Query.prototype.observeOne) {
      Parse.Query.prototype.observeOne = patches.observeOne;
    }
```

And it worked, I could get data from parse!
Thank you very much.

**Comment by dcalogue** [comment]
That's what I did too, but be warned that we are overwriting a Parse function (not only adding a new one, which was what the Patch did before), so some functionality may be missing (I think the function we are overriding is related to the new liveQuery functionality)

**Comment by BrunoAl** [comment]
Someone should really fix this for the next ParseReact release.

**Comment by paulods** [comment]
Is anyone still working on the next ParseReact release?

**Comment by haydenbleasel** [comment]
You can install an earlier version of Parse. I went back to `parse@1.16.4`, which has fixed it.

**Comment by freeznet** [comment]
@haydenbleasel `parse@1.6.14` solved this problem but without `LiveQuery` supporting
